BDArmory/Guidances/MissileGuidance.cs

```diff
@@ -1256,7 +1273,11 @@ public static Vector3 GetAirToAirFireSolution(MissileBase missile, Vector3 targe
                     leadPosition = missile.vessel.CoM + Vector3.RotateTowards(relPos, missile.vessel.upAxis, (theta - angle) * Mathf.Deg2Rad, vertDist);
             }
 
-            return leadPosition;
+            // Don't lead so much you end up leading behind the vehicle...
+            if (Vector3.Dot(targetPosition - missile.vessel.CoM, leadPosition) > 0)
```

Comment:
This is a dot product between a direction and a position. Did you mean `leadPosition - missile.vessel.CoM` here?